src/NHibernate/Linq/Clauses/NhHavingClause.cs

```diff
@@ -1,19 +1,91 @@
-using Remotion.Linq.Clauses;
-using Remotion.Linq.Clauses.ExpressionTreeVisitors;
+using System;
 using System.Linq.Expressions;
+using Remotion.Linq;
+using Remotion.Linq.Clauses;
 
 namespace NHibernate.Linq.Clauses
 {
-	public class NhHavingClause : WhereClause
+	public class NhHavingClause : IBodyClause
 	{
-		public NhHavingClause(Expression predicate) 
-			: base(predicate)
+		Expression _predicate;
+
+		/// <summary>
+		///     Initializes a new instance of the <see cref="T:Remotion.Linq.Clauses.WhereClause" /> class.
+		/// </summary>
+		/// <param name="predicate">The predicate used to filter data items.</param>
+		public NhHavingClause(Expression predicate)
+		{
+			if (predicate == null) throw new ArgumentNullException(nameof(predicate));
+			_predicate = predicate;
+		}
+
+		/// <summary>
+		///     Gets the predicate, the expression representing the where condition by which the data items are filtered
+		/// </summary>
+		public Expression Predicate
+		{
+			get { return _predicate; }
+			set
+			{
+				if (value == null) throw new ArgumentNullException(nameof(value));
+				_predicate = value;
+			}
+		}
+
+		/// <summary>
+		///     Accepts the specified visitor by calling its
+		///     <see
+		///         cref="M:Remotion.Linq.IQueryModelVisitor.VisitWhereClause(Remotion.Linq.Clauses.WhereClause,Remotion.Linq.QueryModel,System.Int32)" />
+		///     method.
+		/// </summary>
+		/// <param name="visitor">The visitor to accept.</param>
+		/// <param name="queryModel">The query model in whose context this clause is visited.</param>
+		/// <param name="index">
+		///     The index of this clause in the <paramref name="queryModel" />'s
+		///     <see cref="P:Remotion.Linq.QueryModel.BodyClauses" /> collection.
+		/// </param>
+		public void Accept(IQueryModelVisitor visitor, QueryModel queryModel, int index)
 		{
+			if (visitor == null) throw new ArgumentNullException(nameof(visitor));
+			if (queryModel == null) throw new ArgumentNullException(nameof(queryModel));
+			((INhQueryModelVisitor) visitor).VisitNhHavingClause(this, queryModel, index);
```

Comment:
Check this out: https://www.re-motion.org/jira/browse/RMLNQ-86. In short they added IBodyClause and IFromClause because of us.